collector/lib/CollectorConfig.cpp

```diff
@@ -447,6 +451,73 @@ void CollectorConfig::HandleConfig(const std::filesystem::path& filePath) {
   }
 }
 
+void WaitForFileToExist(const std::filesystem::path& filePath) {
+  while (!std::filesystem::exists(filePath)) {
+    sleep(5);
+  }
+}
+
+void CollectorConfig::WatchConfigFile(const std::filesystem::path& filePath) {
+  CLOG(INFO) << "In WatchConfigFile";
+  int fd = inotify_init();
+  if (fd < 0) {
+    CLOG(FATAL) << "inotify_init failed";
+    return;
+  }
+
+  WaitForFileToExist(filePath);
```

Comment:
It seems that `inotify` doesn't work unless the file exists to begin with, so I think this is needed.

Reply:
In the future `inotify` will monitor the directory for the creation of the file, so `WaitForFileToExist` will not be needed.